source/vir/src/ast.rs

```diff
@@ -621,8 +620,10 @@ pub enum ExprX {
     Header(HeaderExpr),
     /// Assert or assume
     AssertAssume { is_assume: bool, expr: Expr },
-    /// Forall or assert-by statement; proves "forall vars. ensure" via proof.
-    Forall { vars: Binders<Typ>, require: Expr, ensure: Expr, proof: Expr },
+    /// Assert-forall or assert-by statement; proves "assumption" via proof,
+    /// where "assumption" is "forall vars. require ==> ensure"
+    /// ("assumption" is redundant, but is kept separately because poly.rs treats it differently)
```

Comment:
Can you briefly expand on how it is treated differently? Either here, or in poly.rs.